### DISCUSSION TITLE
Probe offset not working

### DISCUSSION THREAD

**Discussion by Captaindad** [Jyers/Marlin#566] [General Discussion]
I am having an issue where the probe x and y offsets are not doing anything on the screen on my Ender 3 V2.  I change them and store the settings and the offset never changes.  It is causing my head to crash into the right vertical support.  I have reloaded the firmware 3 times as well as restoring factory defaults at least 10 times to no avail.  On V1.3.1.

**Comment by gothcha** [comment]
Hi, under what conditions does the head hit the frame? What firmware version are you using? The XY probe offsets (in advanced) only mainly affect the probe staying on the bed on the left, it will not limit the travel on the right during UBL, the margins are set in the configs, so worst case you may have to compile your own. This said the defaults should be fine unless you have an additional obstruction on the hot end.

**Reply by Captaindad** [reply]
V1.3.1, but I’m not running the UBL, I was running the HS 5x5.  I have an adapter for a “chain guide” on the hot end but on the Smith version .16 and .17 I was running this wasn’t an issue. 

The x y offsets do not move it on the left either. The head stays in the same position.

**Reply by gothcha** [reply]
The XY probe offsets do not move the head as such, just compensates for the BLTouch position.
The problem could well be that to enable as much of the right side of the bed to be probed, Jyers has extended the X travel beyond 230mm, if I recall it is 245mm. The Smith3D I assume will be using 230mm. On the stock V2 245mm is fine but there is only a few mm clearance. Your setup may not have sufficient. I suspect that you will need a custom build with Xmax travel set to 230. Maybe @Jyers can confirm?

**Reply by Captaindad** [reply]
Crap...I do t really know enough about compiling the firmware to do it myself. Is there anyway to change this through the terminal or without recompiling?

**Reply by gothcha** [reply]
No sadly it has to be set in the configuration file, it is very easy to compile your own Marlin version if you use Visual Studio Code, there are lots of tutorials available. Once you get the hang of it you can readily optimise the firmware to your liking.

**Reply by Captaindad** [reply]
I’ll look it up, until then I may go back to the Smith firmware.

**Reply by gothcha** [reply]
When you are ready this is the line you need to edit in configuration.h (1243)

#define X_MAX_POS X_BED_SIZE + 15 // Extended max to allow the probe to reach more of the bed.

If you remove the +15 it will limit the carriage to 230mm

**Reply by Captaindad** [reply]
I’m trying to print a new hotend mount that moves the connection further to the left. If that works I should be good. Thanks for the help.

**Reply by Captaindad** [reply]
It was my mount for sure, I was so used to Smith that I thought it was a firmware issue. 

Thanks.

**Reply by gothcha** [reply]
Fantastic, you will love the firmware! Glad to to able to help.
